### ISSUE TITLE
[BUG] it report that printf function is never invoked but it is invoked

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#3097]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2329
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# shellcheck enable=all

command 1> /dev/null -v 'printf' || {
    printf()
    {
      :
    }
}

printf '%s\n' 'abc'
exit 0
```

#### Here's what shellcheck currently says:

```
    printf()
    ^-- [SC2329] (info): This function is never invoked. Check usage (or ignored if invoked indirectly).
```
#### Here's what I wanted or expected to see:

Nothing
